.github/workflows/build.sh

```diff
@@ -20,7 +24,37 @@ case "$1" in
             export UBSAN_OPTIONS=print_stacktrace=1:print_summary=1:halt_on_error=1
         fi
 
-        ./bootstrap.sh --enable-tests --prefix=/usr
+        if [[ "$WERROR" == true ]]; then
+            # acx_pthread isn't compatible with CFLAGS=-Werror at all so it's just replaced
+            # with ax_pthread from autoconf-archive.
+            rm -rf common/acx_pthread.m4
+
+            export CFLAGS="-Werror"
+
+            # The following warnings are ignored because it's hard to fix them all in one fell swoop.
+            # Ideally they should be fixed one by one and removed from here to let compilers catch them
+            # automatically on PRs.
+            if [[ "$CC" == gcc ]]; then
+                CFLAGS+=" -Wno-error=format-truncation -Wno-error=deprecated-declarations -Wno-error=implicit-fallthrough"
+            elif [[ "$CC" == clang ]]; then
+                CFLAGS+=" -Wno-error=strict-prototypes -Wno-error=invalid-source-encoding -Wno-error=varargs -Wno-error=cast-align"
+                CFLAGS+=" -Wno-error=logical-not-parentheses -Wno-error=deprecated-declarations"
+            fi
+
+            # avahi-gobject triggers a lot of various warnings so it's just easier to turn
+            # it off entirely instead of disabling all the warnings with -Wno-error.
+            additional_autogen_args+=" --disable-gobject"
```

Comment:
Would be nice to fill at least issue that gobject produces a lot of warnings and including issue reference in comment.

Reply:
I started building `avahi-gobject` with `clang` and adjusted the comment accordingly so it's ignored with `gcc` only because all those "deprecated" warnings can't be disabled for some reason with gcc: https://gitlab.gnome.org/GNOME/glib/-/issues/2247. I'm not sure if it makes sense to open an issue covering `avahi-gobject` specifically because all those warnings are visible when `avahi` is built and the script kind of already documents that `avahi` fails to compile with `-Werror` in general and goes even further by listing specific warnings every component triggers with both gcc and clang. For the record `gobject` fails to compile with `gcc` with `-Werror`:
<details>
  <summary>error: Deprecated pre-processor symbol: replace with "G_ADD_PRIVATE"</summary>

```
ga-client.c: In function 'ga_client_init':
ga-client.c:67:13: error: Deprecated pre-processor symbol: replace with "G_ADD_PRIVATE" [-Werror]
   67 |     GaClientPrivate *priv = GA_CLIENT_GET_PRIVATE(self);
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
ga-client.c: In function 'ga_client_set_property':
ga-client.c:81:13: error: Deprecated pre-processor symbol: replace with "G_ADD_PRIVATE" [-Werror]
   81 |     GaClientPrivate *priv = GA_CLIENT_GET_PRIVATE(client);
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
ga-client.c: In function 'ga_client_get_property':
ga-client.c:98:13: error: Deprecated pre-processor symbol: replace with "G_ADD_PRIVATE" [-Werror]
   98 |     GaClientPrivate *priv = GA_CLIENT_GET_PRIVATE(client);
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
ga-client.c: In function 'ga_client_class_init':
ga-client.c:116:5: error: 'g_type_class_add_private' is deprecated [-Werror=deprecated-declarations]
  116 |     g_type_class_add_private(ga_client_class, sizeof (GaClientPrivate));
      |     ^~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/glib-2.0/gobject/gobject.h:24,
                 from /usr/include/glib-2.0/gobject/gbinding.h:29,
                 from /usr/include/glib-2.0/glib-object.h:22,
                 from ga-client.h:23,
                 from ga-client.c:27:
/usr/include/glib-2.0/gobject/gtype.h:1346:10: note: declared here
 1346 | void     g_type_class_add_private       (gpointer                    g_class,
      |          ^~~~~~~~~~~~~~~~~~~~~~~~
ga-client.c: In function 'ga_client_dispose':
ga-client.c:157:13: error: Deprecated pre-processor symbol: replace with "G_ADD_PRIVATE" [-Werror]
  157 |     GaClientPrivate *priv = GA_CLIENT_GET_PRIVATE(self);
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
ga-client.c: In function '_avahi_client_cb':
ga-entry-group.c: In function 'ga_entry_group_init':
ga-entry-group.c:82:13: error: Deprecated pre-processor symbol: replace with "G_ADD_PRIVATE" [-Werror]
   82 |     GaEntryGroupPrivate *priv = GA_ENTRY_GROUP_GET_PRIVATE(obj);
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
ga-client.c:217:13: error: Deprecated pre-processor symbol: replace with "G_ADD_PRIVATE" [-Werror]
  217 |     GaClientPrivate *priv = GA_CLIENT_GET_PRIVATE(self);
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
ga-entry-group.c: In function 'ga_entry_group_get_property':
ga-entry-group.c:99:13: error: Deprecated pre-processor symbol: replace with "G_ADD_PRIVATE" [-Werror]
   99 |     GaEntryGroupPrivate *priv = GA_ENTRY_GROUP_GET_PRIVATE(group);
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
ga-client.c:226:17: error: implicit conversion from 'AvahiClientState' to 'GaClientState' [-Werror=enum-conversion]
  226 |     priv->state = state;
      |                 ^
ga-entry-group.c: In function 'ga_entry_group_class_init':
ga-entry-group.c:115:5: error: 'g_type_class_add_private' is deprecated [-Werror=deprecated-declarations]
  115 |     g_type_class_add_private(ga_entry_group_class,
      |     ^~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/glib-2.0/gobject/gobject.h:24,
                 from /usr/include/glib-2.0/gobject/gbinding.h:29,
                 from /usr/include/glib-2.0/glib-object.h:22,
                 from ga-error.h:23,
                 from ga-entry-group.c:30:
/usr/include/glib-2.0/gobject/gtype.h:1346:10: note: declared here
 1346 | void     g_type_class_add_private       (gpointer                    g_class,
      |          ^~~~~~~~~~~~~~~~~~~~~~~~
ga-entry-group.c: In function 'ga_entry_group_dispose':
ga-entry-group.c:143:13: error: Deprecated pre-processor symbol: replace with "G_ADD_PRIVATE" [-Werror]
  143 |     GaEntryGroupPrivate *priv = GA_ENTRY_GROUP_GET_PRIVATE(self);
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
ga-client.c: In function 'ga_client_start_in_context':
ga-client.c:236:13: error: Deprecated pre-processor symbol: replace with "G_ADD_PRIVATE" [-Werror]
  236 |     GaClientPrivate *priv = GA_CLIENT_GET_PRIVATE(client);
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
ga-entry-group.c: In function 'ga_entry_group_finalize':
ga-entry-group.c:166:13: error: Deprecated pre-processor symbol: replace with "G_ADD_PRIVATE" [-Werror]
  166 |     GaEntryGroupPrivate *priv = GA_ENTRY_GROUP_GET_PRIVATE(self);
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
ga-client.c:248:36: error: implicit conversion from 'GaClientFlags' to 'AvahiClientFlags' [-Werror=enum-conversion]
  248 |                                priv->flags,
      |                                ~~~~^~~~~~~
ga-entry-group.c: In function '_avahi_entry_group_cb':
ga-entry-group.c:215:13: error: Deprecated pre-processor symbol: replace with "G_ADD_PRIVATE" [-Werror]
  215 |     GaEntryGroupPrivate *priv = GA_ENTRY_GROUP_GET_PRIVATE(self);
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
ga-entry-group.c:222:17: error: implicit conversion from 'AvahiEntryGroupState' to 'GaEntryGroupState' [-Werror=enum-conversion]
  222 |     priv->state = state;
      |                 ^
ga-client.c: In function 'ga_client_get_property':
ga-client.c:105:13: error: this statement may fall through [-Werror=implicit-fallthrough=]
  105 |             g_value_set_enum(value, priv->flags);
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
ga-client.c:106:9: note: here
  106 |         default:
      |         ^~~~~~~
ga-entry-group.c: In function 'ga_entry_group_add_service_full_strlist':
ga-entry-group.c:378:13: error: Deprecated pre-processor symbol: replace with "G_ADD_PRIVATE" [-Werror]
  378 |     GaEntryGroupPrivate *priv = GA_ENTRY_GROUP_GET_PRIVATE(group);
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
ga-entry-group.c: In function 'ga_entry_group_add_record_full':
ga-entry-group.c:487:13: error: Deprecated pre-processor symbol: replace with "G_ADD_PRIVATE" [-Werror]
  487 |     GaEntryGroupPrivate *priv = GA_ENTRY_GROUP_GET_PRIVATE(group);
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
ga-entry-group.c: In function 'ga_entry_group_service_thaw':
ga-entry-group.c:517:13: error: Deprecated pre-processor symbol: replace with "G_ADD_PRIVATE" [-Werror]
  517 |             (GA_ENTRY_GROUP_GET_PRIVATE(priv->group)->group,
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
ga-entry-group.c: In function 'ga_entry_group_attach':
ga-entry-group.c:575:13: error: Deprecated pre-processor symbol: replace with "G_ADD_PRIVATE" [-Werror]
  575 |     GaEntryGroupPrivate *priv = GA_ENTRY_GROUP_GET_PRIVATE(group);
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
ga-entry-group.c: In function 'ga_entry_group_commit':
ga-entry-group.c:599:13: error: Deprecated pre-processor symbol: replace with "G_ADD_PRIVATE" [-Werror]
  599 |     GaEntryGroupPrivate *priv = GA_ENTRY_GROUP_GET_PRIVATE(group);
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
ga-entry-group.c: In function 'ga_entry_group_reset':
ga-entry-group.c:614:13: error: Deprecated pre-processor symbol: replace with "G_ADD_PRIVATE" [-Werror]
  614 |     GaEntryGroupPrivate *priv = GA_ENTRY_GROUP_GET_PRIVATE(group);
      |             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
make[3]: *** [Makefile:725: libavahi_gobject_la-ga-client.lo] Error 1
make[3]: *** Waiting for unfinished jobs....
cc1: all warnings being treated as errors
make[3]: *** [Makefile:732: libavahi_gobject_la-ga-entry-group.lo] Error 1
make[3]: Leaving directory '/home/vagrant/avahi/avahi-gobject'
make[2]: *** [Makefile:608: all] Error 2
make[2]: Leaving directory '/home/vagrant/avahi/avahi-gobject'
make[1]: *** [Makefile:831: all-recursive] Error 1
make[1]: Leaving directory '/home/vagrant/avahi'
make: *** [Makefile:741: all] Error 2
```